### ISSUE TITLE
RestTemplate with jetty client leads to IllegalArgumentException with "Self-suppression not permitted"

### ISSUE THREAD

**Issue by ReuDa** [spring-projects/spring-framework#33375]
**Affects:**
- spring-web 6.1.11

There is a bug in the exception handling of the `JettyClientHttpRequest`.

If a RestTemplate is used with the Jetty client and we try to connect to a nonexistent HTTP endpoint, we get an `IllegalArgumentException` with the message `Self-suppression not permitted`.

You can find a reproducer in the [attached zip](https://github.com/user-attachments/files/16600061/reproducer.zip), including a pom.xml. 

```
 assertThatThrownBy(() -> this.restTemplateBuilder.build().exchange("http://localhost:12345", HttpMethod.POST, new HttpEntity<>("body", new HttpHeaders()), String.class))
    .isNotInstanceOf(IllegalArgumentException.class)
    .hasMessageNotContaining("Self-suppression not permitted");
```

**Comment by sdeleuze** [comment]
@ReuDa Thanks for the repro, I tend to think this bug is on Jetty side and looks like matching with https://github.com/jetty/jetty.project/issues/12029. Do you agree?

**Comment by ReuDa** [comment]
@sdeleuze Yes, that seems to be a perfect fit. The described workaround in the linked issue is working fine for me.

Thanks and sorry for the noise!

**Comment by sdeleuze** [comment]
Np, thanks for the quick feedback.

**Comment by icguy** [comment]
FYI this has been fixed in https://github.com/jetty/jetty.project/issues/12029 and shipped with jetty 12.0.26. Which as far as I can tell is used since Spring 6.2.11, so the latest 6.2.x should have this fix.
